### PR TITLE
Add inet:service:object interface to most it:dev:repo forms

### PR DESCRIPTION
**Pull request by invisig0th**

### CODE REVIEW

File: synapse/models/infotech.py [hunk 1]
```diff
@@ -1834,120 +1842,125 @@ def getModelDefs(self):
...
-                    }),
+                        'doc': 'The URL where the branch is hosted.'}),
+
                     ('created', ('time', {}), {
-                        'doc': 'The time this branch was created',
-                    }),
+                        'deprecated': True,
+                        'doc': 'Deprecated. Please use :period.'}),
+
                     ('merged', ('time', {}), {
```
Comment (therealsilence) [inline synapse/models/infotech.py:1862]: we deprecated :created / :deleted to inherit period instead, do we need to deprecate :merged?

File: synapse/models/infotech.py [hunk 2]
```diff
@@ -1834,120 +1842,125 @@ def getModelDefs(self):
...
-                    }),
+                        'doc': 'The text describing the issue.'}),
+
                     ('created', ('time', {}), {
-                        'doc': 'The time the issue was created.',
-                    }),
+                        'deprecated': True,
+                        'doc': 'Deprecated. Please use :period.'}),
+
                     ('updated', ('time', {}), {
```
Comment (therealsilence) [inline synapse/models/infotech.py:1926]: we deprecated :created  to inherit :period instead, do we need to deprecate :updated or is this still relevant?

File: synapse/models/infotech.py [hunk 3]
```diff
@@ -1973,31 +1986,33 @@ def getModelDefs(self):
...
-                    }),
+                        'doc': 'The URL where the comment is hosted.'}),
+
                     ('created', ('time', {}), {
-                        'doc': 'The time the comment was created.',
-                    }),
+                        'deprecated': True,
+                        'doc': 'Deprecated. Please use :period.'}),
+
                     ('updated', ('time', {}), {
```
Comment (therealsilence) [inline synapse/models/infotech.py:2013]: we deprecated :created to inherit period instead, do we need to deprecate :updated or is that still relevant?

File: synapse/models/infotech.py [hunk 4]
```diff
@@ -1834,120 +1842,125 @@ def getModelDefs(self):
...
+
                     ('desc', ('str', {}), {
                         'disp': {'hint': 'text'},
-                        'doc': 'The description of the label.',
-                    }),
+                        'doc': 'The description of the label.'}),
+
                 )),
 
                 ('it:dev:repo:issue:label', {}, (
```
Comment (therealsilence) [inline synapse/models/infotech.py:1950]: Do we need to deprecate :applied / :removed due to inheriting :period?

File: synapse/models/infotech.py [hunk 5]
```diff
@@ -1834,120 +1842,125 @@ def getModelDefs(self):
...
+                        'doc': 'The time the label was applied.'}),
+
                     ('removed', ('time', {}), {
-                        'doc': 'The time the label was removed.',
-                    }),
+                        'doc': 'The time the label was removed.'}),
+
                 )),
 
                 ('it:dev:repo:issue:comment', {}, (
```
Comment (therealsilence) [inline synapse/models/infotech.py:1966]: Do we need to deprecate :created ( and :updated?) due to inheriting :period?

**Review by therealsilence — CHANGES_REQUESTED** [review]
Mostly questions on consistency related to inheriting :period vs. existing time props.

File: synapse/models/infotech.py (hunk 1 above)
Comment (invisig0th) [reply synapse/models/infotech.py:1862]: No, the time when a change was merged does not necessarily mean the branch was closed so it's worth tracking independently.

File: synapse/models/infotech.py (hunk 2 above)
Comment (invisig0th) [reply synapse/models/infotech.py:1926]: still relevant :+1:

File: synapse/models/infotech.py (hunk 4 above)
Comment (invisig0th) [reply synapse/models/infotech.py:1950]: good catch. i missed that one, thanks!

File: synapse/models/infotech.py (hunk 5 above)
Comment (invisig0th) [reply synapse/models/infotech.py:1966]: deprecating `:created` but `:updated` is still relevant. thanks!

File: synapse/models/infotech.py (hunk 3 above)
Comment (invisig0th) [reply synapse/models/infotech.py:2013]: still relevant :+1:

**Review by therealsilence — APPROVED** [review]

File: synapse/models/infotech.py [hunk 6]
```diff
@@ -1834,120 +1842,127 @@ def getModelDefs(self):
...
                     ('id', ('str', {}), {
-                        'doc': 'The version control system specific commit identifier.',
-                    }),
+                        'doc': 'The version control system specific commit identifier.'}),
+
                     ('created', ('time', {}), {
-                        'doc': 'The time the commit was made.',
-                    }),
+                        'deprecated': True,
+                        'doc': 'Deprecated. Please use :period.'}),
```
Comment (Cisphyx) [inline synapse/models/infotech.py:1889]: Using an ival for a commit time feels a little odd but I definitely see the advantage for consistency.

**Review by MichaelSquires — DISMISSED** [review]

**Review by Cisphyx — APPROVED** [review]

File: synapse/models/infotech.py (hunk 6 above)
Comment (invisig0th) [reply synapse/models/infotech.py:1889]: Maybe a commit is more of an atomic `inet:service:action`?

Comment (Cisphyx) [reply synapse/models/infotech.py:1889]: Yeah it feels more like an action than an object

**Review by Cisphyx — APPROVED** [review]

**Review by vEpiphyte — APPROVED** [review]

**Pull request merged**